pkg/controller/federatedcluster/controller.go

```diff
@@ -38,32 +38,44 @@ import (
 	"github.com/golang/glog"
 )
 
+// ClusterController is responsible for maintaining the status of each
+// FederatedCluster in a particular namespace.
 type ClusterController struct {
-	fedClient  fedclientset.Interface
+	// fedClient is used to access Federation resources in the host cluster.
+	fedClient fedclientset.Interface
+	// kubeClient is used to access Secrets in the host cluster.
 	kubeClient kubeclientset.Interface
-	crClient   crclientset.Interface
+	// crClient is used to access the cluster registry in the host cluster.
+	crClient crclientset.Interface
 
 	// clusterMonitorPeriod is the period for updating status of cluster
 	clusterMonitorPeriod time.Duration
 
-	mu              sync.RWMutex
+	mu sync.RWMutex
+
+	// knownClusterSet is the set of clusters known to this controller.
 	knownClusterSet sets.String
-	// clusterClusterStatusMap is a mapping of clusterName and cluster status of last sampling
+
+	// clusterClusterStatusMap is a mapping of clusterName and cluster status of
+	// last sampling.
 	clusterClusterStatusMap map[string]fedv1a1.FederatedClusterStatus
-	// clusterKubeClientMap is a mapping of clusterName and restclient
+
+	// clusterKubeClientMap is a mapping of clusterName and restclient.
 	clusterKubeClientMap map[string]ClusterClient
 
+	// clusterController is the cache.Controller where callbacks are registered
+	// for events on FederatedClusters.
 	clusterController cache.Controller
 
 	// fedNamespace is the name of the namespace containing
-	// FederatedCluster resources and their associated secrets
+	// FederatedCluster resources and their associated secrets.
```

Comment:
Is there a mechanical mechanism to enforce that comments end in a period?

Reply:
I'm sure there's something we can use to enforce it, and I sincerely doubt we want to go there :)

Reply:
I'm not sure of the value of rewriting the docs in a consistent way (end with a period) if we have to manually enforce that consistency going forward.